x-pack/solutions/security/plugins/security_solution/public/common/lib/telemetry/events/siem_migrations/types.ts

```diff
@@ -6,24 +6,78 @@
  */
 
 import type { RootSchema } from '@kbn/core/public';
+import type { RuleMigrationResourceType } from '../../../../../../common/siem_migrations/model/rule_migration.gen';
+import type { SiemMigrationRetryFilter } from '../../../../../../common/siem_migrations/constants';
 
 export enum SiemMigrationsEventTypes {
-  TranslatedRuleUpdate = 'Translated Rule Update',
-  TranslatedRuleInstall = 'Translated Rule Install',
-  TranslatedRuleBulkInstall = 'Translated Rule Bulk Install',
+  SetupConnectorSelected = 'siem_migrations_setup_connector_selected',
+  SetupMigrationOpenNew = 'siem_migrations_setup_rules_migration_open_new',
+  SetupMigrationCreated = 'siem_migrations_setup_rules_migration_created',
+  SetupResourcesUploaded = 'siem_migrations_setup_rules_resources_uploaded',
+  SetupMigrationOpenResources = 'siem_migrations_setup_rules_migration_open_resources',
+  SetupRulesQueryCopied = 'siem_migrations_setup_rules_query_copied',
+  SetupMacrosQueryCopied = 'siem_migrations_setup_rules_macros_query_copied',
+  SetupLookupNameCopied = 'siem_migrations_setup_rules_lookup_name_copied',
+  StartTranslation = 'siem_migrations_start_rules_translation',
+  TranslatedRuleUpdate = 'siem_migrations_translated_rule_update',
+  TranslatedRuleInstall = 'siem_migrations_translated_rule_install',
+  TranslatedRuleBulkInstall = 'siem_migrations_translated_rule_bulk_install',
 }
 
-export interface BaseReportActionParams {
-  migrationId: string;
+export interface BaseResultActionParams {
   result: 'success' | 'failed';
   errorMessage?: string;
 }
 
-export interface ReportTranslatedRuleUpdateActionParams extends BaseReportActionParams {
+// Setup actions
+
+interface ReportSetupConnectorSelectedActionParams {
+  connectorType: string;
+  connectorId: string;
+}
+
+interface ReportSetupMigrationOpenNewActionParams {
+  isFirstMigration: boolean;
+}
+interface ReportSetupMigrationOpenResourcesActionParams {
```

Comment:
super nit, but you could make a `interface BaseSiemParams { migrationId: string; }` and just extend from this for the time being, but not necessary

Reply:
Yes, I tried, but since `migrationId` is optional in some events, and I have the generic type for `result` and `errorMessage`, I decided to define migrationId explicitly in each one.